base/loading.jl

```diff
@@ -220,7 +219,7 @@ function _require_search_from_serialized(node::Int, mod::Symbol, sourcepath::Str
             if isa(restored, ErrorException) && endswith(restored.msg, " uuid did not match cache file.")
                 # can't use this cache due to a module uuid mismatch,
                 # defer reporting error until after trying all of the possible matches
-                failedpath = path_to_try
+                DEBUG_LOADING[] && info(restored, prefix="INFO: JL_DEBUG_LOADING: Failed to load $path_to_try because ")
```

Comment:
that's the prefix, the rest of the error message is `restored.msg` and will complete the thought with: "Module Foo uuid did not match cache file."